lang/csharp/src/apache/main/Schema/LogicalSchema.cs

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using Avro.Util;
+using Newtonsoft.Json.Linq;
+
+namespace Avro
+{
+    /// <summary>
+    /// Class for logical type schemas.
+    /// </summary>
+    public class LogicalSchema : UnnamedSchema
+    {
+        /// <summary>
+        /// The name of the logical type JSON property.
+        /// </summary>
+        public static readonly string LogicalTypeProperty = "logicalType";
+
+        /// <summary>
+        /// Schema for the underlying type that the logical type is based on.
+        /// </summary>
+        public Schema BaseSchema { get; set; }
```

Comment:
It appears these properties are never set outside this class. I think we can safely make them `private set`.

```suggestion
        public Schema BaseSchema { get; private set; }
```